### PR TITLE
update the typeform component so it actually works

### LINKED ISSUE #6241: Typeform embeds don't work for new typeforms.
### Describe the bug

Typeform changed its embed system, so our typeform component no longer works for new typeform pages.

### Details

Our current code:

```html
<script src="https://embed.typeform.com/embed.js" ></script>
<meta name="typeform-formurl" content="https://anil730725.typeform.com/to/LMPh3Tat">
<button id="btn-typeform-popup" class="btn btn-primary">Share your thoughts</button>
```

New "full page" embed code:
```html
<html> <head> <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0"> <title>Political Survey [DEMO 2] (copy)</title> <style type="text/css"> html{ margin: 0; height: 100%; overflow: hidden; } iframe{ position: absolute; left:0; right:0; bottom:0; top:0; border:0; } </style> </head> <body> <iframe id="typeform-full" width="100%" height="100%" frameborder="0" allow="camera; microphone; autoplay; encrypted-media;" src="https://form.typeform.com/to/vjyH5Eum?typeform-medium=embed-snippet"></iframe> <script type="text/javascript" src="https://embed.typeform.com/embed.js"></script> </body> </html>
```

New "standard" embed code:
```html
<div class="typeform-widget" data-url="https://form.typeform.com/to/vjyH5Eum?typeform-medium=embed-snippet" style="width: 100%; height: 500px;"></div> <script> (function() { var qs,js,q,s,d=document, gi=d.getElementById, ce=d.createElement, gt=d.getElementsByTagName, id="typef_orm", b="https://embed.typeform.com/"; if(!gi.call(d,id)) { js=ce.call(d,"script"); js.id=id; js.src=b+"embed.js"; q=gt.call(d,"script")[0]; q.parentNode.insertBefore(js,q) } })() </script>
```

The new embeds also have some design options around width/height and seamless embed:

image

### Things to do

- [x] add support for the new format
- [ ] ~optionally add support for those new design options?~ deferring until necessary
- [ ] ~make sure old embeds still work~ there are no pages that  use typeform in production atm

Addtionally, add `https://form.typeform.com` to `CSP_FRAME_SRC`:

- [x] env.default
- [x] review app
- [x] staging
- [x] production

### Deadlines

This issue is on a deadline, and must be closed out as part of the March 15 sprint.

### PR DESCRIPTION
**Pull request by Pomax**
Closes #6241

Edit examples:
- new type: https://foundation-s-typeform-c-osdpu1.herokuapp.com/cms/pages/33/edit/
- old type: https://foundation-s-typeform-c-osdpu1.herokuapp.com/cms/pages/34/edit/

Live examples:
- https://foundation-s-typeform-c-osdpu1.herokuapp.com/en/campaigns/single-page/
- https://foundation-s-typeform-c-osdpu1.herokuapp.com/en/campaigns/multi-page/

### CODE REVIEW

File: env.default [hunk 1]
```diff
@@ -40,7 +40,7 @@ CSP_CONNECT_SRC="*"
 CSP_DEFAULT_SRC="'none'"
 CSP_FONT_SRC="'self' https://fonts.gstatic.com https://fonts.googleapis.com https://code.cdn.mozilla.net"
 CSP_FRAME_ANCESTORS="'none'"
-CSP_FRAME_SRC="'self' https://www.youtube.com https://comments.mozillafoundation.org/ https://airtable.com https://docs.google.com/ https://platform.twitter.com https://public.zenkit.com https://calendar.google.com https://www.youtube-nocookie.com https://devopstypeform.typeform.com"
+CSP_FRAME_SRC="'self' https://www.youtube.com https://comments.mozillafoundation.org/ https://airtable.com https://docs.google.com/ https://platform.twitter.com https://public.zenkit.com https://calendar.google.com https://www.youtube-nocookie.com https://devopstypeform.typeform.com  https://form.typeform.com"
```
Comment (mmmavis) [inline env.default:43]:
```suggestion
CSP_FRAME_SRC="'self' https://www.youtube.com https://comments.mozillafoundation.org/ https://airtable.com https://docs.google.com/ https://platform.twitter.com https://public.zenkit.com https://calendar.google.com https://www.youtube-nocookie.com https://form.typeform.com"
```

Comment (mmmavis) [reply env.default:43]: Seems like we no longer need to whitelist `https://devopstypeform.typeform.com` for the embed to work

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/typeform_block.html [hunk 2]
```diff
@@ -1,5 +1,7 @@
-<script src="https://embed.typeform.com/embed.js" ></script>
+<div
+  class="typeform-widget"
+  data-url="https://form.typeform.com/to/{{ value.embed_id }}?typeform-medium=embed-snippet"
+  style="width: 100%; height: {{ value.height }}px;">
+</div>
 
-<meta name="typeform-formurl" content="{{ value.url }}">
-
-<button id="btn-typeform-popup" class="btn {{ value.button_type }}">{{ value.button_text }}</button>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/typeform_block.html:5]: From [embed-typeform.js](https://github.com/mozilla/foundation.mozilla.org/blob/master/source/js/embed-typeform.js) it seems like the button was used to launch the form in a new tab. Have we checked with design team to see if it's okay to remove this functionality? If so, let's clean up `embed-typeform.js` a bit.

**Review by mmmavis — CHANGES_REQUESTED** [review]

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/typeform_block.html (hunk 2 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/typeform_block.html:5]: I will turn this into an `<a>` instead, but with button styling, so we don't actually need any JS for this to work.

**Review by sabrinang — COMMENTED** [review]
@Pomax I think previously we had just the button without the embeded form above it. With all the form elements and button, it is looking a bit visually busy too if they are both present.

I believe the button launched the form as a popup that is still on the current page (like a [lightbox](https://d1nt04ao0vck22.cloudfront.net/uploads/2015/07/15090400/header-footer-enabled.png)), is that still a possibility with the updated typeform component?

**Comment by Pomax** [comment]
@sabrinang @mmmavis it looks like the makePopup is still available, so I'll restore the original code and only update the component a little to use the new URL format.

**Comment by Pomax** [comment]
@mmmavis @sabrinang I've updated the PR to use the "popup" logic again.

**Review by sabrinang — APPROVED** [review]
Awesome, thanks for making the pop-up work Pomax!

**Comment by Pomax** [comment]
@mmmavis this is ready for code review

File: source/js/embed-typeform.js [hunk 3]
```diff
@@ -9,27 +9,33 @@ const EmbedTypeform = {
...
+
+    const popup = window.typeformEmbed.makePopup(typeformLink.href, {
+      mode: "popup",
+      hideHeaders: true,
+      hideFooter: true,
+      autoOpen: false,
+    });
+
+    const open = () => popup.open();
+    ["touchstart", "click"].forEach((evtType) =>
```
Comment (mmmavis) [inline source/js/embed-typeform.js:36]: touchstart 👍

**Review by mmmavis — APPROVED** [review]
Woohooo works great!

**Pull request merged**